folding-schemes/src/commitment/hyrax.rs

```diff
@@ -0,0 +1,253 @@
+    }
+}
+
+fn eq_evals<F: Field>(point: &[F]) -> Vec<F> {
+    let n = 1 << point.len();
+    cfg_into_iter!(0..n)
+        .map(|i| {
+            let mut eval = F::one();
+            for (j, &bit) in point.iter().enumerate() {
+                let b = (i >> j) & 1 == 1;
+                eval *= if b { bit } else { F::one() - bit };
+            }
+            eval
+        })
+        .collect()
+}
+
+fn matrix_dimensions(num_vars: usize, _ratio: usize) -> (usize, usize) {
+    let row = (num_vars / 2).next_power_of_two();
+    let col = (1 << num_vars) / row;
+    (col, row)
+}
+
+impl<C: Curve, const H: bool> CommitmentScheme<C, H> for Hyrax<C> {
+    type ProverParams = HyraxGenerators<C>;
+    type VerifierParams = HyraxGenerators<C>;
+    type Proof = Proof<C>;
+    type ProverChallenge = ();
+    type Challenge = ();
+
+    fn is_hiding() -> bool {
+        false
+    }
+
+    fn setup(
+        mut rng: impl RngCore,
+        len: usize,
+    ) -> Result<(Self::ProverParams, Self::VerifierParams), Error> {
+        let (col_len, row_len) = matrix_dimensions(len, 1);
+        let pedersen_generators = (0..row_len).map(|_| C::Affine::rand(&mut rng)).collect();
+        let params = HyraxGenerators {
+            pedersen_generators,
+            row_len,
+            col_len,
+        };
+        Ok((params.clone(), params))
+    }
+
+    fn commit(
+        params: &Self::ProverParams,
+        v: &[C::ScalarField],
+        _r: &C::ScalarField,
+    ) -> Result<C, Error> {
+        if v.len() != params.row_len * params.col_len {
+            return Err(Error::PedersenParamsLen(
+                params.row_len * params.col_len,
+                v.len(),
+            ));
+        }
+
+        let commitment = Hyrax::commit(v, params)?;
+        Ok(commitment
+            .row_commitments
+            .iter()
+            .copied()
+            .reduce(|a, b| a + b)
+            .unwrap_or_else(C::zero))
+    }
+
+    fn prove(
+        params: &Self::ProverParams,
+        _transcript: &mut impl Transcript<C::ScalarField>,
+        _cm: &C,
+        v: &[C::ScalarField],
+        _r: &C::ScalarField,
+        _rng: Option<&mut dyn RngCore>,
+    ) -> Result<Self::Proof, Error> {
+        let point = vec![C::ScalarField::zero(); (params.col_len as f64).log2() as usize];
+        let evaluation = Hyrax::open(v, &point, params)?;
+        Ok(Proof { evaluation })
+    }
+
+    fn prove_with_challenge(
+        _params: &Self::ProverParams,
+        _challenge: Self::ProverChallenge,
+        _v: &[C::ScalarField],
+        _r: &C::ScalarField,
+        _rng: Option<&mut dyn RngCore>,
+    ) -> Result<Self::Proof, Error> {
+        unreachable!("Hyrax does not use prove_with_challenge")
+    }
+
+    fn verify(
+        _params: &Self::VerifierParams,
+        _transcript: &mut impl Transcript<C::ScalarField>,
+        _cm: &C,
+        _proof: &Self::Proof,
+    ) -> Result<(), Error> {
+        todo!("Verification not yet implemented")
```

Comment:
How difficult it would be to implement verify on a different PR given the fact that you already implemented the open method?